src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+"""
+:created: 2024-02-02
+:author: Nandish Patel
+:copyright: © 2024 Nandish Patel, Delmic
+
+This file is part of Odemis.
+
+.. license::
+    Odemis is free software: you can redistribute it and/or modify it under the
+    terms of the GNU General Public License version 2 as published by the Free
+    Software Foundation.
+
+    Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+    PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along with
+    Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+from abc import ABCMeta, abstractmethod
+
+import wx
+
+from odemis import model, util
+from odemis.acq.stream import UNDEFINED_ROI
+from odemis.gui.comp.overlay.base import WorldOverlay
+
+UNDEFINED_POS_SIZE = (0, 0)
+
+
+class EditableShape(metaclass=ABCMeta):
+    """
+    This abstract EditableShape class forms the base for a series of classes that
+    refer to shape tools like rectangle, ellipse, polygon and their functionality.
+    """
+
+    def __init__(self, cnvs):
+        """:param cnvs: canvas passed by the shape's overlay and used to draw the shapes."""
+        # States if the shape is selected
+        # The VA's value should always be set using _set_value(selected, must_notify=True)
+        # method because the points and/or coordinates should always be updated id the shape
+        # was selected
+        self.selected = model.BooleanVA(True)
+        # list of nested points (y, x) representing the shape and whose value will be used
+        # during ROA acquisition
+        # FastEMROA.get_poly_field_indices expects list of nested tuples (y, x)
+        self.points = model.ListVA()
+        # Any shape can be represented by a bounding box and the coordinates (l, t, r, b) are
+        # stored here
+        self.coordinates = model.TupleVA(UNDEFINED_ROI)
```

Comment:
Is this VA to be used *in addition* to `.points`, or instead of it?
Couldn't we just have `.points` and automatically compute `.coordinates` when needed? Also, `coordinates` doesn't really convey the meaning it's a single rectangle. I'd suggest `.boundingBox` (as we use in other parts of Odemis). Or just prove a .getBoundingBox() on the class, which automatically computes it based on the .points VA.